src/compiler/watchUtilities.ts

```diff
@@ -365,12 +365,17 @@ export function createCachedDirectoryStructureHost(host: DirectoryStructureHost,
     }
 }
 
-/** @internal */
-export enum ConfigFileProgramReloadLevel {
-    None,
-    /** Update the file name list from the disk */
-    Partial,
-    /** Reload completely by re-reading contents of config file from disk and updating program */
+export enum ProgramUpdateLevel {
+    /** Program is updated with same root file names and options */
+    Update,
+    /** Loads program after updating root file names from the disk */
+    RootNamesAndUpdate,
+    /**
+     * Loads program completely, including:
+     *   re-reading contents of config file from disk
+     *   calculating root file names for the program
+     *   Updating the program
+     */

```

Comment:
Nit: depending on the markdown-ish interpretation of the editor, these lines may have been rendered separated only by a space, which would be confusing. This will either make it parse as a real list or at least separate the items with a dash.

```suggestion
     * Loads program completely, including:
     *  - re-reading contents of config file from disk
     *  - calculating root file names for the program
     *  - Updating the program
     */

```